### ISSUE TITLE
Suggestion - Some machines should have 'upgrade' slots that allow to insert various items to affect the machine's modifiers

### ISSUE THREAD

**Issue by Nextej** [loiste-interactive/Obenseuer-Issues#2600]
Machines should have a functionality of having additional slots (between 1-4 I guess) that the player can insert various designated items into. As the things stand currently, I see at least few cases where addition of such would be beneficial.

- Fluid containers to have at least 1 slot that the player can use to insert things like:
  - Funnel (increase filling speed)
  - Piece of Rubber Hose (allow unfilling the barrels, which otherwise should be only possible to be filled for storage)
  - Small Electric Pump (same as above, just with increased unfilling speed)
- Stoves should have 2/4 slots that player can use to insert things like kettles (for tea), frying pan (for various recipes) and pots (for various recipes) that are required to start the crafting
- Sawbuck and Sawbuck with Chainsaw should be combined into singular machine with 1 slot that player can insert either a Saw or a Chainsaw.

**Comment by LeLastpak** [comment]
Yeah some machines are begging to be upgradeable and it would be nice to be able to be customized according to the needs of the player.

Things that the player want to upgrade are:
 - production speed
 - production capacity(able to craft multiple items at once)
 - volume capacity(distillers really would want this for bigger jobs)
 - input slots(atm its often maxed at 9 but sometimes you need more, for 100L, mashing jobs for example)
 - quality output increase
 - gas upgrade(runs on gas instead of fuel)
 - electricity upgrade(runs on electricity instead of fuel)
